concept/type_/type_manager.rs

```diff
@@ -743,62 +742,62 @@ impl<Snapshot: WritableSnapshot> TypeManager<Snapshot> {
     }
 }
 
-pub trait ReadableType<'a, 'b>: TypeAPI<'a> + CommonType<'a> {
+pub trait ReadableType {
     // Consider replacing 'b with 'static
-    type SelfRead: ReadableType<'b, 'b>;
-    fn read_from(b: Bytes<'b, BUFFER_KEY_INLINE>) -> Self::SelfRead;
+    type Output<'out>: 'out;
+    fn read_from<'out>(b: Bytes<'out, BUFFER_KEY_INLINE>) -> Self::Output<'out>;
 }
```

Comment:
`'b` or `'bytes` is better at communicating which lifetime the output is tied to.